packages/web/components/Dashboard/Calendar.tsx

```diff
@@ -1,33 +1,470 @@
-import { Flex, Text } from '@metafam/ds';
-import { CONFIG } from 'config';
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  CalendarAddIcon,
+  CalendarIcon,
+  ExternalLinkIcon,
+  Flex,
+  HStack,
+  IconButton,
+  Image,
+  LoadingState,
+  MetaButton,
+  Popover,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverFooter,
+  PopoverHeader,
+  PopoverTrigger,
+  Portal,
+  Text,
+  Tooltip,
+  VStack,
+} from '@metafam/ds';
+import { MetaLink } from 'components/Link';
+import { MarkdownViewer } from 'components/MarkdownViewer';
+import type { GoogleCalEventType } from 'lib/hooks/useCalendar';
+import { useCalendar } from 'lib/hooks/useCalendar';
+import { DateTime } from 'luxon';
 import React, { useEffect, useState } from 'react';
 
+import { calendarID } from './config';
+
+type GroupedEventsType = {
+  date: string;
+  events: GoogleCalEventType[];
+};
+
 export const Calendar: React.FC = () => {
-  const [isComponentMounted, setIsComponentMounted] = useState(false);
+  const [calendar, setCalendar] = useState<GroupedEventsType[]>([]);
+  const [limit, setLimit] = useState(4);
+  const { events, fetching, error } = useCalendar(limit);
+  const calICS = `https://calendar.google.com/calendar/ical/${calendarID}%40group.calendar.google.com/public/basic.ics`;
+  // const usersOffset = DateTime.local().offset / 60;
+  // const usersOffsetString = usersOffset > 0 ? `+${usersOffset}` : usersOffset;
 
-  useEffect(() => setIsComponentMounted(true), []);
+  // const getUrlStringFromDescription = (desc: string) => {
+  //   const regex = /(?<=")[^"]+(?=")/g;
+  //   const cover = desc ? desc.match(regex) : '';
 
-  if (!isComponentMounted) {
-    return null;
-  }
+  //   return cover;
+  // }
+
+  // strip out the +++cover+++ from the description
+  const cleanDescription = (desc: string) =>
+    desc ? desc.replace(/(\+\+\+).*(\+\+\+)/, '') : '';
+  const buildAddToCalendarLink = (event: GoogleCalEventType) => {
+    const start =
+      'dateTime' in event.start
+        ? DateTime.fromISO(event.start.dateTime)
+        : DateTime.fromISO(event.start.date);
+    const end =
+      'dateTime' in event.end
+        ? DateTime.fromISO(event.end.dateTime)
+        : DateTime.fromISO(event.end.date);
+    const title = event.summary;
+    const { location } = event;
+    const details = `${cleanDescription(event.description)}`;
+    const dates = `${start.toFormat('yyyyMMdd')}T${start.toFormat(
+      'HHmmss',
+    )}/${end.toFormat('yyyyMMdd')}T${end.toFormat('HHmmss')}`;
+    const href = `https://www.google.com/calendar/render?action=TEMPLATE&text=${title}&dates=${dates}&details=${details}&location=${location}&sf=true&output=xml`;
+    return href;
+  };
+
+  const groupEventsByDay = (items: GoogleCalEventType[]) => {
```

Comment:
ditto, I think this should be in the backend code, from that you can just return a nice UI-friendly set of data that can be just rendered